src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java

```diff
@@ -86,6 +82,39 @@ public ValidationResult check(Entry entry) throws ValidationEngineException {
         return result;
     }
     
+    private String getDivisionByTaxid(SourceFeature primarySF){
+        String division = "";
+        if(primarySF.getTaxId()!=null && getDivisionFromCache(primarySF.getTaxId().toString())!=null){
+            division = getDivisionFromCache(primarySF.getTaxId().toString());
+        }else {
+            division = new TaxonomyClientImpl().getTaxonByTaxid(primarySF.getTaxId()).getDivision();
+            saveDivisionCache(primarySF.getTaxId().toString(), division);
+        }
+        return division;
+    }
+    
+    private String getDivisionByScientificName(SourceFeature primarySF){
+        String division = "";
+        if(primarySF.getScientificName()!=null && getDivisionFromCache(primarySF.getScientificName())!=null){
+            division = getDivisionFromCache(primarySF.getScientificName());
+        }else {
+            List<Taxon> taxonList;
+            if (!(taxonList = new TaxonomyClientImpl().getTaxonByScientificName(primarySF.getScientificName())).isEmpty()) {
+                division = taxonList.get(0).getDivision();
+                saveDivisionCache(primarySF.getScientificName(), division);
+            }
+        }
+        return division;
+    }
+    
+    private String getDivisionFromCache(String divisionKey){
+        return EmblEntryValidationPlan.divisionCache.get(divisionKey)!=null ?  EmblEntryValidationPlan.divisionCache.get(divisionKey) : null;
+    }
+    
+    private void saveDivisionCache(String divisionKey, String division){
```

Comment:
- Minor name change suggestion: divisionKey -> scientificName

Reply:
Name suggestion (divisionKey -> scientificName) is not added as the same method is used for used for taxId and scientificName